core_api/src/dependencies.py

```diff
@@ -60,7 +72,59 @@ async def get_vector_store(
     )
 
 
-async def get_llm(env: Annotated[Settings, Depends(get_env)]) -> ChatLiteLLM:
+class ESQuery(TypedDict):
+    question: str
+    file_uuids: list[UUID]
+    user_uuid: UUID
+
+
+def get_es_retriever(
```

Comment:
Not sure, there are more changes coming on the dependency stuff but I don't think it will help here. Right now I'm thinking we might be able to do https://python.langchain.com/v0.1/docs/expression_language/primitives/configure/. That might tie up well with the injected admin values George mentioned, a chain is then invoked with a ChainInput and a set of config values?

Reply:
TIL, Thanks for pointing this out